pkg/install/generate_test.go

```diff
@@ -0,0 +1,93 @@
+package install
+
+import (
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1alpha1"
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"testing"
+)
+
+const (
+	testClusterName       = "bar"
+	sshKeySecret          = "ssh-key"
+	pullSecretSecret      = "pull-secret"
+)
+
+func init() {
+	log.SetLevel(log.DebugLevel)
+}
+
+func TestGenerate(t *testing.T) {
+	tests := []struct {
+		name      string
+		expectErr bool
+		validate  func(*testing.T)
+	}{
+		{
+			name: "Testing PreserveOnDelete",
+			validate: func(t *testing.T) {
+				cd := testClusterDeployment()
+				cd.Spec.PreserveOnDelete = true
```

Comment:
Technically you only need these 5 lines for the test, the struct helps set us up for adding more tests easily in future but doing the assertion inside validate here makes me think we should probably wipe out lines 25-32 and 38-47. validate() would be for "post execution" validation whereas in this case it's actually the whole test, so here I think we can just go even simpler. expectErr was also unused.

Otherwise looks great, thanks!